holidayextras/ui-toolkit
# UXUI-210 Creating an INPUT component

#### What does this PR do? (please provide any background)
- The work was initially produced by Peter and I have adjusted slightly and created a new PR (a lot of noise on the last one)
- This creates an input component which handles `text, tel, number, email` with more to come
- Includes onClick events, required and some validation

#### What tests does this PR have?
run `npm test` and it should run all mocha tests. we have included new ones for this component

#### How can be this tested?
run `grunt docs` to see this on the UXUI toolkit docs, have a play with the live code editor.

#### Screenshots / Screencast
![screenshot_2015-07-15-11-20-06_720](https://cloud.githubusercontent.com/assets/778942/8696379/b9d9412c-2ae3-11e5-9c77-d830029e3912.png)
![screenshot_2015-07-15-11-20-01_720](https://cloud.githubusercontent.com/assets/778942/8696382/c2177750-2ae3-11e5-90ba-28df6830bcc8.png)


#### What gif best describes how you feel about this work?
![Angelos](http://i.imgur.com/a8HGS7Z.gif)

---
#### Developer Definition of Done/Quality Checklist (for PR author to complete BEFORE code review):

#### Software Engineer or Developer review:

#### Software Engineer or project guru review:
